### PR TITLE
Improve docs on Swift version support.

### PR DESCRIPTION
**Pull request by thomasvl**
- Spell out the Xcode version it maps too.
- Document what should happen when changing the min Swift version.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,9 +100,10 @@ your project as explained below.
 
 To use Swift with Protocol buffers, you'll need:
 
-* A recent Swift 3 compiler that includes the Swift Package Manager.  The Swift
-protobuf project is being developed and tested against the latest release version of
-Swift available from [Swift.org](https://swift.org)
+* A recent Swift 3 compiler (Xcode 8.x or later).  Support is included for the Swift
```
Comment (tbkka) [inline README.md]: Should we be more specific here?

    (we currently require Swift 3.0.1 or later, as provided by Xcode 8.1 or later)

File: Documentation/INTERNALS.md [hunk 2]
```diff
@@ -15,6 +15,19 @@ As a result, this document is probably already out of date;
 pull requests that correct this document to better match the actual
 behavior are always appreciated.
 
+## Swift Language Support
```
Comment (tbkka) [inline Documentation/INTERNALS.md]: Looks good.

File: README.md (hunk 1 above)
Comment (thomasvl) [inline README.md]: Tweaked

File: Documentation/INTERNALS.md [hunk 3]
```diff
@@ -15,6 +15,19 @@ As a result, this document is probably already out of date;
...
+
+The goal is to always support "one full major version”. Which basically
```
Comment (allevato) [inline Documentation/INTERNALS.md]: Nit: s/. Which/, which/

File: Documentation/INTERNALS.md [hunk 4]
```diff
@@ -15,6 +15,19 @@ As a result, this document is probably already out of date;
...
+means if the current official release of Swift is `X.Y`, the library will
+support back to `X-1.Y`.  I.E.- when Swift 4.1 gets released, the minimum
```
Comment (allevato) [inline Documentation/INTERNALS.md]: Nit: s/I.E.-/That is, /

File: Documentation/INTERNALS.md (hunk 3 above)
Comment (thomasvl) [inline Documentation/INTERNALS.md]: Done

File: Documentation/INTERNALS.md (hunk 4 above)
Comment (thomasvl) [inline Documentation/INTERNALS.md]: Done

**Pull request merged**